src/main/java/com/powsybl/networkconverterserver/NetworkConverterService.java

```diff
@@ -0,0 +1,122 @@
+import com.powsybl.networkconverterserver.dto.NetworkIds;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.network.store.client.NetworkStoreClient;
+import com.powsybl.network.store.client.NetworkStoreService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.DefaultUriBuilderFactory;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
+
+@Service
+@ComponentScan(basePackageClasses = {NetworkStoreClient.class})
+public class NetworkConverterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkConverterService.class);
+    private String caseServerBaseUri;
+    private String geoDataBaseUri;
+    private RestTemplate caseServerRest;
+    private RestTemplate geoDataRest;
+    private static final  String IIDM_GEO_DATA_API_VERSION = "v1";
+
+    @Autowired
+    private NetworkStoreService networkStoreService;
+
+    @Autowired
+    public NetworkConverterService(@Value("${backing-services.case.base-uri:http://case-server/}") String caseServerBaseUri,
+                                   @Value("${backing-services.geo-data.base-uri:http://geo-data-server/}") String geoDataBaseUri) {
+        RestTemplateBuilder restTemplateBuilder = new RestTemplateBuilder();
+        caseServerRest = restTemplateBuilder.build();
+        caseServerRest.setUriTemplateHandler(new DefaultUriBuilderFactory(caseServerBaseUri));
+        this.caseServerBaseUri = caseServerBaseUri;
+
+        restTemplateBuilder = new RestTemplateBuilder();
+        geoDataRest = restTemplateBuilder.build();
+        geoDataRest.setUriTemplateHandler(new DefaultUriBuilderFactory(geoDataBaseUri));
+        this.geoDataBaseUri = geoDataBaseUri;
+    }
+
+    NetworkIds persistentStore(String caseName) {
+        byte[] networkByte = getCaseAsByte(caseName);
+        String[] baseName = caseName.split("\\.");
+        ReadOnlyMemDataSource readOnlyMemDataSource = new ReadOnlyMemDataSource(baseName[0]);
+        readOnlyMemDataSource.putData(caseName, networkByte);
+        Network network = networkStoreService.importNetwork(readOnlyMemDataSource);
+        UUID networkUuid = networkStoreService.getNetworkUuid(network);
+        precalculateLines(networkUuid);
+        return new NetworkIds(networkUuid, network.getId());
+    }
+
+    private void precalculateLines(UUID idNetwork) {
```

Comment:
What is it?

Reply:
@ChamseddineBhd you'll probably explain this better than me

Reply:
@geofjamg  we used this method before to order lines coordinates when importing a new network.
@AbdelHedhili you can delete it's useless now!